### DISCUSSION TITLE
Allow the tags to work with uuids

### DISCUSSION THREAD

**Discussion by jeroendesloovere** [forkcms/forkcms#3343] [Ideas / Feature Requests]
## Type

Enhancement

## Problem description

You can now only use integers as id's for the tags, if switched to strings we can use UUID's

## Why?

When creating modules with **Doctrine entities and UUID**,
$id is `char(36)` in combination with `(DC2Type:guid)`.

So using the current tags module will not be possible anymore for these custom modules,
since it requires the $otherId to be an `int` and we have a `char`...
```
// save the tags
BackendTagsModel::saveTags(int $otherId, $tags, string $module, string $language = null)
```

## Related issue with same problem
"Search to work with uuids": https://github.com/forkcms/forkcms/issues/1962

**Comment by tijsverkoyen** [comment]
Can you explain why? And an example on why you need this?

**Comment by jeroendesloovere** [comment]
## Why?

When creating modules with **Doctrine entities and UUID**,
$id is `char(36)` in combination with `(DC2Type:guid)`.

So using the current tags module will not be possible anymore for these custom modules,
since it requires the $otherId to be an `int` and we have a `char`...
```
// save the tags
BackendTagsModel::saveTags(int $otherId, $tags, string $module, string $language = null)
```

## Related issue with same problem
"Search to work with uuids": https://github.com/forkcms/forkcms/issues/1962

**Comment by carakas** [comment]
@tijsverkoyen I've also ran into this issue, and this combined with the search module is the main reason I'm not using them more often. Since if you want that to work you need to add a separate id just for this. Kinda overkill

**Comment by carakas** [comment]
@jeroendesloovere can you update the issue description so it contains the reasons why that you added in your comment?

**Comment by carakas** [comment]
For now you can fix it with something like this

Entity
```php
    /**
     * @var string
     *
     * @ORM\Id()
     * @ORM\GeneratedValue(strategy="UUID")
     * @ORM\Column(type="guid")
     */
    private $id;

    /**
     * @var int
     *
     * we need this since the tag index only works with integers
     *
     * @ORM\Column(type="integer")
     */
    private $tagId;

    /**
     * @param int $tagId
     *   …
     */
    public function __construct(
        $tagId
        …
    ) {
```

Creating new instance of entity

```php
        $myEntity = new MyEntity(
            $this->myEntityRepository->getNextTagId(),
            …
        );
```

Repository
```php
    public function getNextTagId(): int
    {
        return (int) $this->getEntityManager()->createQueryBuilder()
            ->select('(COALESCE(MAX(e.tagId), 0) + 1) AS tagId')
            ->from(MyEntity::class, 'e')
            ->getQuery()
            ->getSingleScalarResult();
    }
```
